code/modules/security/workshop.dm

```diff
@@ -43,8 +42,7 @@
 	return data
 
 /obj/machinery/computer/holodeck/prison/ui_act(action, params)
-	if(!allowed(usr))
-		to_chat(usr, span_warning("Access denied."))
+	if(isobserver(usr))
```

Comment:
Check for allowed (because that checks for things, such as admin AI interaction), but change the req_access of the machine rather than checking for isobserver.